### PR TITLE
WFP-752 added storing previous conviction date

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/jpa/entity/UnallocatedCaseEntity.kt [hunk 1]
```diff
@@ -34,5 +34,8 @@ data class UnallocatedCaseEntity(
   val initial_appointment: LocalDate? = null,
 
   @Column
-  val status: String
+  val status: String,
+
+  @Column
+  val previous_conviction_date: LocalDate? = null
```
Comment (trozzomoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/jpa/entity/UnallocatedCaseEntity.kt:40]: Not sure on why we use the null initialisation for entity properties. Can we just declare  LocalDate?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UnallocatedCasesService.kt [hunk 2]
```diff
@@ -43,15 +43,21 @@ class UnallocatedCasesService(
...
+        val allConvictions = communityApiClient.getAllConvictions(crn)
         return when {
-          allConvictions > 1 -> "Previously managed"
-          else -> "New to probation"
+          allConvictions.size > 1 -> {
+            val mostRecentInactiveConvictionEndDate =
+              allConvictions.filter { c -> !c.active && c.sentence.terminationDate != null }
+                .map { c -> c.sentence.terminationDate!! }
+                .maxByOrNull { it }
+            ProbationStatus("Previously managed", mostRecentInactiveConvictionEndDate)
```
Comment (trozzomoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UnallocatedCasesService.kt:58]: status should not be managed in enum or static properties somewhere ?

**Review by markreesmoj — APPROVED** [review]
🥇

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UnallocatedCasesService.kt [hunk 3]
```diff
@@ -43,15 +43,21 @@ class UnallocatedCasesService(
     return hmppsTierApiClient.getTierByCrn(crn)
   }
 
-  fun getProbationStatus(crn: String): String {
+  fun getProbationStatus(crn: String): ProbationStatus {
     val activeConvictions = communityApiClient.getActiveConvictions(crn).size
     return when {
-      activeConvictions > 1 -> "Currently managed"
```
Comment (trozzomoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UnallocatedCasesService.kt:49]: Not sure about this when construct. We have some logic based on size on 2 different place that I guess should be encapusalte din functions with a menaingfulName that describe why >1 is relevant .

**Pull request merged**